src/script/script.cpp

```diff
@@ -227,3 +238,33 @@ bool CScript::IsPushOnly() const
     }
     return true;
 }
+
+int CScript::Type() const
```

Comment:
I'm not the biggest fan of the `Type()` method, but I do think it is an improvement on the code that this is coming from. I considered what if this returned an enum, or what if the function return a pair of type address hash (to avoid the other method), or even to put in some polymorphic structure to encapsulate all of this. That being said said, I'm not sure that any of these suggestions are worth it given that it is fine as is.

Reply:
@Eirik0 After further thought, I decided to leave this as-is; it would be disruptive to change it to an enum or make other kinds of changes.